crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -194,10 +190,16 @@ impl Trade {
             order::FeePolicy::Volume { factor } => Ok(self.volume_fee(*factor)?.amount),
         };
 
-        let protocol_fee = self.policies.first().map(protocol_fee).transpose();
+        let amount = self
+            .policies
+            .iter()
+            .try_fold(TokenAmount::default(), |acc, protocol_fee| {
+                acc.checked_add(&calc_protocol_fee(protocol_fee).ok()?)
+            })
+            .unwrap_or_default();
```

Comment:
This seems slightly sketchy (was already before). To get better introspection here we should log a warning when we actually fallback to 0 due to some error.